### ISSUE TITLE
[Bug]: Silent renew using access token not id token like documented

### ISSUE THREAD

**Issue by spikehockey75** [damienbod/angular-auth-oidc-client#1731]
### Version

15

### Please provide the exception or error you saw

```true
[DEBUG] 0-scfr_ui - silent renew, periodic check finished!
angular-auth-oidc-client.mjs:4249 [DEBUG] 0-scfr_ui - Checking: silentRenewRunning: false, isCodeFlowInProgress: false - has idToken: true - has userData: true
angular-auth-oidc-client.mjs:4249 [DEBUG] 0-scfr_ui - Has accessToken expired: false --> expires in 8:27 , 11:40:25 AM > 11:31:58 AM
```

### A clear and concise description of what you expected to happen.

```true
We expect the library to refresh tokens on id_token expire time not access_token.
```

### Additional context

We have been using the library for awhile and it used to refresh when id token expired, like silent renew says it does but now it appears to be checking access_token and since they are out of sync its causing issues with our implementation.

**Comment by damienbod** [comment]
Hi @spikehockey75 

We added a configuration for this:

https://angular-auth-oidc-client.com/docs/documentation/configuration#disableidtokenvalidation

Can you check if this is enabled? 

If this is enabled, then it should work,  otherwise we have a bug in the lib.

Greetings Damien

**Comment by spikehockey75** [comment]
we tried manually setting it to false and still does not appear to be working

**Comment by damienbod** [comment]
@spikehockey75 This could be a bug then, can you send your client configuration?

Greetings Damien

**Comment by spikehockey75** [comment]
sure thing!

`        config: {
          authority: environment.stsServer,
          redirectUrl: `${window.location.origin}/resolve`,
          clientId: 'custom_ui',
          scope: 'openid StandardWorkforceAndCDGroups',
          responseType: 'code',
          triggerAuthorizationResultEvent: true,
          postLogoutRedirectUri: `${window.location.origin}/unauthorized`,
          startCheckSession: false,
          silentRenew: true,
          postLoginRoute: '/main',
          ignoreNonceAfterRefresh: true,
          disableRefreshIdTokenAuthTimeValidation: true,
          triggerRefreshWhenIdTokenExpired: true,
          forbiddenRoute: '/forbidden',
          unauthorizedRoute: '/unauthorized',
          useRefreshToken: true,
          logLevel: LogLevel.Debug,
          disableIdTokenValidation: false,
          renewTimeBeforeTokenExpiresInSeconds: 10,
          issValidationOff: true,
          customParamsAuthRequest: {['auth_groups']: 'test_'}
        }
      })],`

**Comment by spikehockey75** [comment]
Any update on this?

**Comment by spikehockey75** [comment]
@damienbod any idea on this?

**Comment by spikehockey75** [comment]
@damienbod  any idea?

**Comment by damienbod** [comment]
@spikehockey75 busy with the day job, will try to make time to look at this

Greetings Damien

**Comment by damienbod** [comment]
Set this property to 

```
disableRefreshIdTokenAuthTimeValidation: false,
```

**Comment by damienbod** [comment]
Just created a PR to fix this, will release it once the tests are good

@spikehockey75  Thanks for reporting

greetings Damien

**Comment by damienbod** [comment]
Will be released in 15.0.5
